enterprise_subsidy/apps/transaction/management/commands/write_reversals_from_enterprise_unenrollments.py

```diff
@@ -0,0 +1,216 @@
+        course_start_date = None
+        for run in content_metadata.get('course_runs'):
+            if run.get('key') == enrollment_course_run_key:
+                course_start_date = run.get('start')
+                break
+
+        if not course_start_date:
+            logger.warning(
+                f"No course start date found for course run: {enrollment_course_run_key}. "
+                "Unable to determine refundability."
+            )
+            return False
+
+        # https://2u-internal.atlassian.net/browse/ENT-6825
+        # OCM course refundability is defined as True IFF:
+        # ie MAX(enterprise enrollment created at, course start date) + 14 days > unenrolled_at date
+        enrollment_created_at = enterprise_course_enrollment.get("created")
+        enrollment_unenrolled_at = enterprise_course_enrollment.get("unenrolled_at")
+        enrollment_created_datetime = datetime.strptime(
+            enrollment_created_at, "%Y-%m-%dT%H:%M:%SZ"
+        )
+        course_start_datetime = datetime.strptime(
+            course_start_date, "%Y-%m-%dT%H:%M:%SZ"
+        )
+        enrollment_unenrolled_at_datetime = datetime.strptime(
+            enrollment_unenrolled_at, "%Y-%m-%dT%H:%M:%SZ"
+        )
+        refund_cutoff_date = max(course_start_datetime, enrollment_created_datetime) + timedelta(days=14)
+
+        if refund_cutoff_date > enrollment_unenrolled_at_datetime:
+            logger.info(
+                f"Course run: {enrollment_course_run_key} is refundable for enterprise customer user: "
+                f"{enterprise_course_enrollment.get('enterprise_customer_user')}. Writing Reversal record."
+            )
+            return True
+        else:
+            logger.info(
+                f"Unenrollment from course: {enrollment_course_run_key} by user: "
+                f"{enterprise_course_enrollment.get('enterprise_customer_user')} is not refundable."
+            )
+            return False
+
+    def handle(self, *args, **options):
+        """
+        Fetch enterprise enrollment objects that have been unenrolled within the last 24 hours and write Transaction
+        Reversals where appropriate for each unenrollment.
+        """
+        DRY_RUN_LOG_PREFIX = ""
+        if dry_run := options.get('dry_run'):
+            DRY_RUN_LOG_PREFIX = "DRY RUN: "
+            logger.info("Running in dry-run mode. No updates or creates will occur.")
+
+        logger.info(
+            f"{DRY_RUN_LOG_PREFIX}Updating and writing Transaction Reversals from recent Enterprise unenrollments data"
+        )
+        recent_unenrollments = EnterpriseApiClient().fetch_recent_unenrollments()
+        logger.info(
+            f"{DRY_RUN_LOG_PREFIX}Found {len(recent_unenrollments)} recent Enterprise unenrollments"
+        )
+
+        reversals_processed = 0
+        fetched_content_metadata = {}
+        for unenrollment in recent_unenrollments:
```

Comment:
might be nice to take the body of this `for` loop and make a method out of it, like `handle_unenrollment(unenrollment)`

Reply:
This can also make the code easier to test

Reply:
Done!